inspirehep/modules/workflows/utils.py

```diff
@@ -83,3 +89,55 @@ def log_workflows_action(action, relevance_prediction,
         }
         audit = WorkflowsAudit(**logging_info)
         audit.save()
+
+
+def with_debug_logging(func):
+    """Generate a debug log with info on what's going to run.
+
+    It tries its best to use the logging facilities of the object passed or the
+    application context before falling back to the python logging facility.
+    """
+    @wraps(func)
+    def _decorator(*args, **kwargs):
+        def _get_obj(args, kwargs):
+            if args:
+                obj = args[0]
+            else:
+                obj = kwargs.get('obj', kwargs.get('record'))
+            return obj
+
+        def _get_logfn(args, kwargs):
+            obj = _get_obj(args, kwargs)
+            if hasattr(obj, 'log') and hasattr(obj.log, 'debug'):
+                logfn = obj.log.debug
+            elif hasattr(current_app, 'log'):
+                logfn = current_app.log.debug
+            else:
+                logfn = LOGGER.debug
+
+            return logfn
+
+        def _proof_run(func, *args, **kwargs):
+            try:
+                func(*args, **kwargs)
+            except Exception:
+                LOGGER.debug(
+                    "Error while trying to log with %s:\n%s",
+                    logfn,
```

Comment:
Magically both of them are the same... but yes, I agree that it should try to use the parameter one instead of the one in the closure. Probably will rename the parameter too, as it collides with the func from the outer decorator.

Maybe I just should move the 'helper' funcs that don't use any closures outside of the function...